relentless/potential/potential.py

```diff
@@ -34,28 +131,123 @@ def evaluate(self, pair):
+            The name of the file to save data in.
+
+        """
+        data = {}
         for key in self:
-            all_params[str(key)] = self.evaluate(key)
+            all_params = {}
+            for param in self[key]:
+                var = self[key][param]
+                if isinstance(var, Variable):
+                    all_params[param] = {'type':'variable', 'value':var.value,
+                                         'const':var.const, 'low':var.low, 'high':var.high}
+                elif isinstance(var, (float,int)):
+                    all_params[param] = {'type':'scalar', 'value':var}
+            data[str(key)] = all_params
         with open(filename, 'w') as f:
-            json.dump(all_params, f, sort_keys=True, indent=4)
+            json.dump(data, f, sort_keys=True, indent=4)
 
     def load(self, filename):
-        raise NotImplementedError('Load is not implemented')
+        """Loads the data from a file into the coefficient matrix.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Raises
+        ------
+        KeyError
+            If the pairs in the data are not exactly the pairs in self
+        KeyError
+            If the params in the data are not exactly the params in self
+        TypeError
+            If the parameter value type in the data and in self do not match
+
+        """
+        with open(filename, 'r') as f:
+            data = json.load(f)
+        pairs = []
+        for key in data:
+            pairs.append(eval(key))
+        if sorted(self.pairs) != sorted(tuple(pairs)):
+            raise KeyError('The type pairs in the data are not exactly the pairs in self')
+        for key in data:
+            key_tup = eval(key)
+            if sorted(self.params) != sorted(data[key].keys()):
+                raise KeyError('The parameters in the data are not exactly the parameters in self')
+            for param in data[key]:
+                var_data = data[key][param]
+                var_self = self[key_tup][param]
+                if var_self is None:
+                    if var_data['type'] == 'variable':
+                        self[key_tup][param] = Variable(value=var_data['value'], const=var_data['const'],
+                                                        low=var_data['low'], high=var_data['high'])
+                    elif var_data['type'] == 'scalar':
+                        self[key_tup][param] = var_data['value']
+                    else:
+                        raise TypeError('The parameter value types in the data and self do not match')
+                elif var_data['type'] == 'scalar' and isinstance(var_self, (float,int)):
+                    self[key_tup][param] = var_data['value']
+                elif var_data['type'] == 'variable' and isinstance(var_self, Variable):
+                    self[key_tup][param] = Variable(value=var_data['value'], const=var_data['const'],
```

Comment:
We don't want to overwrite the reference to the original variable, so instead set everything by property:
```py
self[key][param].value = var_data['value']
```